src/System.ComponentModel.TypeConverter/src/System/ComponentModel/CultureInfoConverter.cs

```diff
@@ -0,0 +1,651 @@
+                    // where the propgrid will send values from this returned array directly -- instead
+                    // of converting it to a string and then back to a value (which this relied on).
+                    array[0] = CultureInfo.InvariantCulture; //null isn't the value here -- invariantculture is.
+                }
+
+                _values = new StandardValuesCollection(array);
+            }
+
+            return _values;
+        }
+
+        /// <summary>
+        ///    <para>
+        ///       Gets a value indicating whether the list of standard values returned from
+        ///       System.ComponentModel.CultureInfoConverter.GetStandardValues is an exclusive list.
+        ///    </para>
+        /// </summary>
+        public override bool GetStandardValuesExclusive(ITypeDescriptorContext context)
+        {
+            return false;
+        }
+
+        /// <summary>
+        ///    <para>
+        ///       Gets a value indicating whether this object supports a
+        ///       standard set of values that can be picked from a list using the specified
+        ///       context.
+        ///    </para>
+        /// </summary>
+        public override bool GetStandardValuesSupported(ITypeDescriptorContext context)
+        {
+            return true;
+        }
+
+        /// <summary>
+        ///      IComparer object used for sorting CultureInfos
+        ///      WARNING:  If you change where null is positioned, then you must fix CultureConverter.GetStandardValues!
+        /// </summary>
+        private class CultureComparer : IComparer
+        {
+            private CultureInfoConverter _converter;
+
+            public CultureComparer(CultureInfoConverter cultureConverter)
+            {
+                Debug.Assert(cultureConverter != null);
+                _converter = cultureConverter;
+            }
+
+            public int Compare(object item1, object item2)
+            {
+                if (item1 == null)
+                {
+                    // If both are null, then they are equal
+                    //
+                    if (item2 == null)
+                    {
+                        return 0;
+                    }
+
+                    // Otherwise, item1 is null, but item2 is valid (greater)
+                    //
+                    return -1;
+                }
+
+                if (item2 == null)
+                {
+                    // item2 is null, so item 1 is greater
+                    //
+                    return 1;
+                }
+
+                String itemName1 = _converter.GetCultureName(((CultureInfo)item1));
+                String itemName2 = _converter.GetCultureName(((CultureInfo)item2));
+
+                CompareInfo compInfo = (CultureInfo.CurrentCulture).CompareInfo;
+                return compInfo.Compare(itemName1, itemName2, CompareOptions.StringSort);
+            }
+        }
+
+        private static class CultureInfoMapper
```

Comment:
This could be improved...

Benefits:
 - Avoids intermediate allocations as items are added to the dictionary by specifying the capacity
 - Avoids looking up the key twice by using `TryGetValue`
 - Avoids manual lazy initialization
 - Allows `s_cultureInfoNameMap` to be `readonly`

```c#
private static class CultureInfoMapper
{
    // Dictionary of CultureInfo.DisplayName, CultureInfo.Name for cultures that have changed DisplayName over releases.
    // This is to workaround an issue with CultureInfoConverter that serializes DisplayName (fixing it would introduce breaking changes).
    private static readonly Dictionary<string, string> s_cultureInfoNameMap = CreateMap();

    private static Dictionary<string, string> CreateMap()
    {
        const int Count = 274;
        var result = new Dictionary<string, string>(Count)
        {
            { "Afrikaans", "af" },
            ...
        };
        Debug.Assert(result.Count == Count);
        return result;
    }

    public static string GetCultureInfoName(string cultureInfoDisplayName)
    {
        string name;
        return s_cultureInfoNameMap.TryGetValue(cultureInfoDisplayName, out name) ?
            name :
            cultureInfoDisplayName;
    }
}
```